src/adapter/base.cr

```diff
@@ -8,22 +8,10 @@ abstract class Granite::Adapter::Base
   property database : DB::Database
 
   def initialize(adapter : String)
-    if url = ENV["DATABASE_URL"]? || Granite.settings.database_url || replace_env_vars(settings(adapter)["database"].to_s)
+    if url = Granite::Settings.database_url[adapter]
```

Comment:
I think this is a good solution to the problem at hand, but I also think it can be made backwards compatible with a little more effort.

Is it possible to retain the _setter_ `Granite::Settings.database_url = 'postgres://u:p@h:5432/db'` syntax and make this happen behind the scenes? That would:

- not require a breaking-change release
- allow people to specify multiple adapter connection strings if needed (though only up to 3, so no connections to multiple postgres instances)

Reply:
I think that is a good solution.  Few things to note:
* If I provide three postgres strings, the last one would be the one that is used since they would just reset the previous ones?
* We would have to either rename some of adapter classes to match the connection string adapter, i.e. `pg -> postgres and sqlite -> sqlite3`, or have conditional logic to handle the change in keys

Reply:
@robacarp I disagree about making this backward compatible instead why not make the `Granite::Settings` can happen at a later PR where there are more breaking changes for a given release. This means that we would have to keep track of the backward compatibility code move any settings property to both places (more duplication) to later remove the dead code.